ateam_common/src/km_assignment.cpp

```diff
@@ -88,6 +87,25 @@ Eigen::MatrixXd replace_nan_costs_with_value(
   return new_matrix;
 }
 
+Eigen::MatrixXd replace_forbidden_costs_with_zeros(
```

Comment:
How is this intended to work?
As I read this, you would be zero-ing out the costs of all tasks for a given robot or vice versa.
But we want to be able to disallow specific robots for specific tasks while allowing those robots to be accurately costed for other tasks.
Also, is there code somewhere that checks for zero cost assignments, or would this still assign disallowed robots if no other bots were available?

Reply:
I think this new implementation is more logical, let me know if you see any clear errors with it.

I think there does need to be more robust checking to see if a particular cost is zero and explicitly not assign those tasks... I need to add some test cases that purposefully fail due to lack of assignable tasks/robots and handle them; right now, I think the assignment function will continue endlessly if there is no actual solution.

Reply:
When we're doing min cost assignment and the cost matrix gets scaled, wouldn't the originally max cost robot-assignment pair be set to zero? If we go forward with "zero cost means disallowed", then that robot would become disallowed unexpectedly.
Is my understanding here correct?

Reply:
Actually looks like I misread what the change was.
I think the above issue might still be a problem? Though I'm also not sure what the role of the code below that sets entries in `xy` to -1 is.